### ISSUE TITLE
getRendererState?xyz problem if Spotify stream

### ISSUE THREAD

**Issue by staedelis** [ChriD/node-raumserver#96]
my raumfeld devices are updated with firmware 2.12.5
the node-raumserver is running on a raspberry PI

getRendererState?xyz does works as long as TuneIN stream is set but it does not work any more in case a spotify stream is set. This was not the case before the firmware update.
For some reason the id is not found. 

In case of the Spotify stream the answer is:

{
requestUrl: "/raumserver/data/getRendererState?id=Schlafzimmer",
action: "getRendererState",
error: true,
msg: "Action was rejected",
data: {
errorMessage: "Error: Media Renderer for id 'Schlafzimmer' not found",
errorData: null
}
}

However the device is visible with getRendererState in the Data array, devices with TuneIn stream have 2 entries, devices with Spotify stream only 1 entry.

example of the existing entry:
{
InstanceID: 0,
AVTransportURIMetaData: "<DIDL-Lite xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/" xmlns:raumfeld="urn:schemas-raumfeld-com:meta-data/raumfeld" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/"><item id="0/Spotify/Track" parentID="0/Spotify" restricted="0"><upnp:class>object.item.audioItem.musicTrack</upnp:class><raumfeld:section>Spotify</raumfeld:section><dc:title>Happy - From "Despicable Me 2"</dc:title><upnp:artist>Pharrell Williams</upnp:artist><upnp:album>G I R L</upnp:album><upnp:albumArtURI dlna:profileID="JPEG_TN">https://i.scdn.co/image/ab67616d0000b273e8107e6d9214baa81bb79bba</upnp:albumArtURI><res duration="0:03:52.000" protocolInfo="spotify:*:audio/spotify-track:*">spotify:track:60nZcImufyMA1MKQY3dcCH</res></item></DIDL-Lite>",
CurrentTrackDuration: "0:03:53",
PowerState: "ACTIVE",
AVTransportURI: "spotify://playback",
CurrentPlayMode: "SHUFFLE",
TransportState: "PAUSED_PLAYBACK",
OwnsAudioResource: "0",
CurrentTransportActions: "Play,Previous,Next,Seek,RepeatTrack,Repeat,Shuffle",
TransportStatus: "OK",
LowDB: "0.000000",
Mute: "0",
MidDB: "0.000000",
Volume: "21",
HighDB: "0.000000",
SettingValue: "1",
rooms: [
{
name: "Kueche-Wohnzimmer",
udn: "uuid:d7a8103d-66d9-4dc3-979d-9adf338c86a6",
powerState: "ACTIVE",
renderer: [
{
udn: "uuid:e25dbcf7-a3fc-4f61-8892-cd32a2748ab9",
name: "Speaker Kueche-Wohnzimmer",
spotifyConnect: "active"
}
]
}
],
udn: "uuid:e25dbcf7-a3fc-4f61-8892-cd32a2748ab9",
mediaItem: null,
friendlyName: "Speaker Kueche-Wohnzimmer",
host: "192.168.178.33",
manufacturer: "Lautsprecher Teufel GmbH",
modelNumber: "1"
},
{
InstanceID: 0,
AVTransportURIMetaData: "<?xml version="1.0"?> <DIDL-Lite xmlns="urn:schemas-upnp-org:metadata-1-0/DIDL-Lite/" xmlns:raumfeld="urn:schemas-raumfeld-com:meta-data/raumfeld" xmlns:upnp="urn:schemas-upnp-org:metadata-1-0/upnp/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dlna="urn:schemas-dlna-org:metadata-1-0/" xmlns:pv="http://www.pv.com/pvns/" lang="en"><item parentID="0/Favorites/RecentlyPlayed" id="0/Favorites/RecentlyPlayed/99" restricted="1" refID="0/RadioTime/Search/s-s297919"><raumfeld:name>Station</raumfeld:name><upnp:class>object.item.audioItem.audioBroadcast.radio</upnp:class><raumfeld:section>RadioTime</raumfeld:section><raumfeld:durability>120</raumfeld:durability><dc:title>101 Smooth Jazz</dc:title><upnp:albumArtURI dlna:profileID="JPEG_TN">http://d1i6vahw24eb07.cloudfront.net/s297919q.png</upnp:albumArtURI><raumfeld:ebrowse>http://opml.radiotime.com/Tune.ashx?partnerId=7aJ9pvV5&amp;formats=mp3%2Cogg&amp;serial=50%3A1e%3A2d%3A31%3Aca%3A90&amp;id=s297919&amp;c=ebrowse</raumfeld:ebrowse></item></DIDL-Lite> ",
CurrentTrackDuration: "0:00:00",
PowerState: "ACTIVE",
AVTransportURI: "http://192.168.178.30:60107/17338c5c-01fc-477d-b409-b550031912e5/0945ce51-9922-443a-bb00-7dc1cbd3813e/26cdc987-2ad6-478a-bc93-fca3a64663f6/d5452524b6575e5151eae5c39fbb4315--545067590-0",
CurrentPlayMode: "NORMAL",
TransportState: "PLAYING",
OwnsAudioResource: "1",
CurrentTransportActions: "Stop",
TransportStatus: "OK",
LowDB: "0.000000",
Mute: "0",
MidDB: "0.000000",
Volume: "13",
HighDB: "0.000000",
SettingValue: "1",
rooms: [
{
name: "Schlafzimmer",
udn: "uuid:0945ce51-9922-443a-bb00-7dc1cbd3813e",
powerState: "ACTIVE",
renderer: [
{
udn: "uuid:26cdc987-2ad6-478a-bc93-fca3a64663f6",
name: "Speaker Schlafzimmer"
}
]
}
],
udn: "uuid:26cdc987-2ad6-478a-bc93-fca3a64663f6",
mediaItem: null,
friendlyName: "Speaker Schlafzimmer",
host: "192.168.178.48",
manufacturer: "Lautsprecher Teufel GmbH",
modelNumber: "1"
}

**Comment by gahujipo** [comment]
I guess you already tried to kill node-raumserver and restart it?

**Comment by staedelis** [comment]
Yes of course and it does not help. The problem is fully reproducible. Stream from TuneIn it works, switch to Spotify stream it doesn't, switch back to TuneIn it works again.

**Comment by staedelis** [comment]
I did some further testing and figured out that it works if in the Raumfeld app the feature multi room for spotify is turned on.

**Comment by fschuber** [comment]
I face the same issue. But as I use the free spotify, I cannot turn on multi room for spotify.
It did work in the past. I'm not sure, if it did stop when I started using spotify.

**Comment by sajov** [comment]
@staedelis
any news on this?

**Comment by webbrain-one** [comment]
I opened PR #106 to fix this issue. It makes `getRendererState` handle Spotify streams correctly by resolving the renderer properly instead of returning “Media Renderer for id not found.”
